pkg/apiserver/clusterinfo/hostinfo/cluster_config.go

```diff
@@ -37,9 +40,44 @@ func FillInstances(db *gorm.DB, m InfoMap) error {
 		if _, ok := m[hostname]; !ok {
 			m[hostname] = NewHostInfo(hostname)
 		}
-		m[hostname].Instances[row.Instance] = &InstanceInfo{
-			Type:           row.Type,
-			PartitionPathL: strings.ToLower(locateInstanceMountPartition(row.Value, m[hostname].Partitions)),
+		switch row.Type {
+		case "tiflash":
+			if ins, ok := m[hostname].Instances[row.Instance]; ok {
+				if ins.Type == row.Type && ins.PartitionPathL != "" {
+					continue
+				}
+			} else {
+				m[hostname].Instances[row.Instance] = &InstanceInfo{
+					Type:           row.Type,
+					PartitionPathL: "",
+				}
+			}
+			var paths []string
+			switch row.Key {
+			case "engine-store.path":
+				items := strings.Split(row.Value, ",")
+				for _, path := range items {
+					paths = append(paths, strings.TrimSpace(path))
+				}
+			case "engine-store.storage.main.dir", "engine-store.storage.latest.dir":
+				if err := json.Unmarshal([]byte(row.Value), &paths); err != nil {
+					return err
+				}
```

Comment:
"engine-store.storage.main.dir", "engine-store.storage.latest.dir" These two configs can be set simultaneously. In this case, we need to merge the two array to `paths`